src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -546,6 +546,14 @@ protected void VisitLvalue(BoundExpression node)
                     ((BoundTupleExpression)node).VisitAllElements((x, self) => self.VisitLvalue(x), this);
                     break;
 
+                case BoundKind.DiscardExpression:
```

Comment:
Please add validation at the compiler layer as well. Let me know if you need pointers for that.

Reply:
I'm not sure I understand what you mean so some pointers would be great 😄

Reply:
The IDE layer (Workspaces, Editor, EditorFeature) rests on top of the Compilers layer (CodeAnalysis, CodeAnalysis.CSharp) which exposes a number of APIs (`Compilation`, `SemanticModel`, `IOperation`, etc). The compiler layer has its own set of tests (for example CSharp.Emit or CSharp.Semantic). 

A change in the Compilers layer must be accompanied by a test change also in the Compilers layer to demonstrate the change.

I took a closer look at this PR to figure out which compiler API is involved in this IDE scenario. So I copied your two new C# tests into Rolsyn from `master` branch and ran them. That would help me trace the link between the IDE scenario and your code change in `AbstractFlowPass` and identify the public compiler API that is involved.

But, to my surprise, both tests passed. So I do not know why a compiler change is necessary or what is fixed by this PR.

Reply:
I figured out what happened. I started working on this forgetting to get latest master so I worked on an older version. I tested that Nullable had the same issue as the given test case and solved that issue. After syncing with master it appears that nullable no longer has the same issue but the original issue is still there.
This PR is currently a noop at best. I'll figure out the real solution and update it 😄